### DISCUSSION TITLE
okhttp asynchronous request mode does not support collection http status

### DISCUSSION THREAD

**Discussion by LeePui** [apache/skywalking#8120] [Q&A]
skywalking java agent version :  8.9.0-SNAPSHOT
    i found in [synchronous call mode](https://github.com/apache/skywalking-java/blob/main/apm-sniffer/apm-sdk-plugin/okhttp-common/src/main/java/org/apache/skywalking/apm/plugin/okhttp/common/RealCallInterceptor.java) afterMethod，it will determine span `erroreoccurred` and set `http.status_code` according to the status code。
    but in [asynchronous call mode](https://github.com/apache/skywalking-java/blob/main/apm-sniffer/apm-sdk-plugin/okhttp-common/src/main/java/org/apache/skywalking/apm/plugin/okhttp/common/AsyncCallInterceptor.java) afterMethod, it just stop the span. in agent version 8.5.0, it will set `erroreoccurred` in [OnFailureInterceptor](https://github.com/apache/skywalking-java/blob/main/apm-sniffer/apm-sdk-plugin/okhttp-common/src/main/java/org/apache/skywalking/apm/plugin/okhttp/common/OnFailureInterceptor.java) and [OnResponseInterceptor](https://github.com/apache/skywalking-java/blob/main/apm-sniffer/apm-sdk-plugin/okhttp-common/src/main/java/org/apache/skywalking/apm/plugin/okhttp/common/OnResponseInterceptor.java ), should we add http status code collection in asynchronous call mode?

**Comment by wu-sheng** [comment]
If you could read error code, then, it is better to collect the response code for all cases.
